code/modules/surgery/organs/subtypes/machine.dm

```diff
@@ -117,37 +117,25 @@
 	robotize()
 	..()
 
-/obj/item/organ/internal/optical_sensor
+/obj/item/organ/internal/eyes/optical_sensor
```

Comment:
IPCs already get screenblinding flash when they weld, been that way for a long time. Plus I think the damage protection is linked to the IPC's head.